src/ape/managers/chain.py

```diff
@@ -1196,6 +1202,44 @@ def instance_at(
             else:
                 raise  # Current exception
 
+        if abi:
+            # if abi is type string then convert it to json object
+            if isinstance(abi, str):
+                try:
+                    # convert the abi from string to list
+                    list_abi = json.loads(abi)
+                    # Create a new ContractType with the provided ABI
+                    contract_type = ContractType(abi=list_abi)
+                except Exception as err:
+                    if contract_type:
+                        # If a default contract type was provided, don't error and use it.
+                        logger.error(str(err))
+                    else:
+                        raise  # Current exception
+            # if it's of type List[ABI] pass it directly to ContractType Class
+            elif isinstance(abi, list):
+                # Use the provided abi directly
+                contract_type = ContractType(abi=abi)
+            elif isinstance(abi, Path):
+                # Handle both absolute and relative paths
+                if not abi.is_absolute():
+                    project_folder = Path.cwd()
+                    abi = project_folder / abi
+                try:
+                    with abi.open() as f:
+                        json_str = f.read()
+                        list_abi = json.loads(json_str)
```

Comment:
You could avoid all this `with` block with a simple:

```python
list_abi = json.loads(abi.read_text())
```

(a trick I learned from @fubuloubu back in the day)